### PR TITLE
Initial working commit

### PR DESCRIPTION
**Pull request by chrisghill**
Pretty simple change that touched a lot of files. Pulling out the `afero` mockfilesystem stuff. It was useful in testing, but some of our dependencies (`airlock`) perform filesystem access which makes it very difficult to maintain. There was really only 3 categories of changes here:

* removing filesystem arguments from function signatures
* changing `afero` filesystem calls (`ReadFile`, `WriteFile`, etc) to their corresponding functions in `os` or `filepath`
* updating tests to perform their actions in a `TempDir`.

### CODE REVIEW

**Comment by chrisghill** [comment]
Only remaining question is what to call the `mockfilesystem` package now that it isn't really a mock. The functions all remain, but they are intended to be used within a `testing.TempDir`.

I thought about `testing`, but that conflicts with the stdlib `testing` package. Perhaps `testfilesystem`, or `testingutils` but I don't like `utils` in packages.

Open to thoughts/suggestion.

**Review by xpositivityx — APPROVED** [review]
This looks good!

File: pkg/bundle/prompt.go [hunk 1]
```diff
@@ -107,8 +106,7 @@ func getDescription(t *templatecache.TemplateData) error {
 var ignoredTemplateDirs = map[string]bool{"alpha": true}
 
 func getTemplate(t *templatecache.TemplateData) error {
-	var fs = afero.NewOsFs()
-	cache, _ := templatecache.NewBundleTemplateCache(templatecache.GithubTemplatesFetcher, fs)
+	cache, _ := templatecache.NewBundleTemplateCache(templatecache.GithubTemplatesFetcher)
```
Comment (likelymichael) [inline pkg/bundle/prompt.go:109]: there's a lint recommendation below, won't allow me to comment on it

File: pkg/bundle/write_schemas.go [hunk 2]
```diff
@@ -39,7 +38,7 @@ func (b *Bundle) WriteSchemas(buildPath string, fs afero.Fs) error {
 
 		filepath := fmt.Sprintf("/schema-%s.json", task.label)
 
-		err = afero.WriteFile(fs, path.Join(buildPath, filepath), content, 0755)
+		err = os.WriteFile(path.Join(buildPath, filepath), content, 0755)
```
Comment (likelymichael) [inline pkg/bundle/write_schemas.go]: would change perms to `0600` or add lint exclusion like `//nolint:gosec` to L17

File: pkg/commands/generate_new_bundle_test.go [hunk 3]
```diff
@@ -10,23 +11,20 @@ import (
...
-	"github.com/spf13/afero"
 	"sigs.k8s.io/yaml"
 )
 
 func TestCopyFilesFromTemplateToCurrentDirectory(t *testing.T) {
-	rootTemplateDir := "/home/md-cloud"
-	writePath := "."
+	testDir := t.TempDir()
+	rootTemplateDir := path.Join(testDir, "/home/md-cloud")
+	writePath := testDir
```
Comment (likelymichael) [inline pkg/commands/generate_new_bundle_test.go]: would look nicer to remove L20 and replace all `writePath` to `testDir`, or replace `testDir` with `writePath`

File: pkg/server/bundle/bundle.go [hunk 4]
```diff
@@ -302,7 +300,7 @@ func (h *Handler) postConnections(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err = afero.WriteFile(h.fs, path.Join(h.bundleDir, "src", bundle.ConnsFile), bytes, 0755)
+	err = os.WriteFile(path.Join(h.bundleDir, "src", bundle.ConnsFile), bytes, 0755)
```
Comment (likelymichael) [inline pkg/server/bundle/bundle.go]: same comment about lint as above

File: pkg/terraform/generate_files.go [hunk 5]
```diff
@@ -83,7 +83,7 @@ func generateTfVarsFiles(buildPath, stepPath string, b *bundle.Bundle, fs afero.
 		}
 
 		filePath := fmt.Sprintf("/_%s_variables.tf.json", task.label)
-		err = afero.WriteFile(fs, path.Join(buildPath, stepPath, filePath), content, 0755)
+		err = os.WriteFile(path.Join(buildPath, stepPath, filePath), content, 0755)
```
Comment (likelymichael) [inline pkg/terraform/generate_files.go]: same as above

File: pkg/terraform/transpile_connection_var_files.go [hunk 6]
```diff
@@ -3,16 +3,16 @@ package terraform
...
-	"github.com/spf13/afero"
 )
 
-func transpileConnectionVarFile(path string, b *bundle.Bundle, fs afero.Fs) error {
+func transpileConnectionVarFile(path string, b *bundle.Bundle) error {
 	emptyConnections := checkEmptySchema(b.Connections)
 
 	if emptyConnections {
-		err := afero.WriteFile(fs, path, []byte("{}"), 0755)
+		err := os.WriteFile(path, []byte("{}"), 0755)
```
Comment (likelymichael) [inline pkg/terraform/transpile_connection_var_files.go]: same as above

File: pkg/terraform/transpile_connection_var_files.go [hunk 7]
```diff
@@ -40,7 +40,7 @@ func transpileConnectionVarFile(path string, b *bundle.Bundle, fs afero.Fs) erro
 		return err
 	}
 
-	err = afero.WriteFile(fs, path, bytes, 0755)
+	err = os.WriteFile(path, bytes, 0755)
```
Comment (likelymichael) [inline pkg/terraform/transpile_connection_var_files.go]: same as above

File: pkg/terraform/transpile_dev_param_var_files.go [hunk 8]
```diff
@@ -3,16 +3,16 @@ package terraform
...
-	"github.com/spf13/afero"
 )
 
-func transpileAndWriteDevParams(path string, b *bundle.Bundle, fs afero.Fs) error {
+func transpileAndWriteDevParams(path string, b *bundle.Bundle) error {
 	emptyParams := checkEmptySchema(b.Params)
 
 	if emptyParams {
-		err := afero.WriteFile(fs, path, []byte("{}"), 0755)
+		err := os.WriteFile(path, []byte("{}"), 0755)
```
Comment (likelymichael) [inline pkg/terraform/transpile_dev_param_var_files.go]: same

File: pkg/terraform/transpile_dev_param_var_files.go [hunk 9]
```diff
@@ -59,7 +59,7 @@ func transpileAndWriteDevParams(path string, b *bundle.Bundle, fs afero.Fs) erro
 		return err
 	}
 
-	err = afero.WriteFile(fs, path, bytes, 0755)
+	err = os.WriteFile(path, bytes, 0755)
```
Comment (likelymichael) [inline pkg/terraform/transpile_dev_param_var_files.go]: same

**Review by mclacore — APPROVED** [review]
Looks good! Approved. Left some suggestions though

**Pull request merged**